src/github.com/getlantern/peerscanner/web.go

```diff
@@ -149,12 +149,29 @@ func clientIpFor(req *http.Request, name string) string {
 	ip := strings.TrimSpace(ips[0])
 	// TODO: need a more robust way to determine when a non-fallback host looks
 	// like a fallback.
-	if !isFallback(name) && strings.HasPrefix(ip, "128.199") {
+	hasFallbackIp := isFallbackIp(ip)
+	if !isFallback(name) && hasFallbackIp {
 		log.Errorf("Found fallback ip %v for non-fallback host %v", ip, name)
 		return ""
-	} else if isFallback(name) && !strings.HasPrefix(ip, "128.199") {
+	} else if isFallback(name) && !hasFallbackIp {
 		log.Errorf("Found non-fallback ip %v for fallback host %v", ip, name)
 		return ""
 	}
 	return ip
 }
+
+func isFallbackIp(ip string) bool {
+	for _, prefix := range fallbackIPPrefixes {
+		if strings.HasPrefix(ip, prefix) {
+			return true
+		}
+	}
+
+	return false
+}
+
+var fallbackIPPrefixes = []string{
+	"128.199",
+	"178.62",
+	"188.166",
+}
```

Comment:
> Dumb question: are we sure that no peer could have an IP matching this?  Or is this just some extra sanity check? 

Great question actually.  This is not a certain check, hence my TODO comment in the code.

> If we want to be positive about whether a server is a fallback, we could have it prove so upon registering, e.g. by providing a secret token that we deploy to our servers.

Yes, that would be one way to make it more robust.

> Something would need to be done for peerscanner startup to decide what to do in the gap until servers first register.  One option is that on startup peerscanner just asks all servers with fallback-like names to register, and it doesn't go into the main loop until all have done so or there's some timeout.

We're already in a state where servers are registered.  On startup, peerscanner pulls all that information in from CloudFlare and as long as the already registered servers pass the test, they remain in rotation.

> Another option is that we just treat servers with fallback-like names as fallbacks until they first register.

We already treat servers with fallback-like names as fallbacks, except if they don't have a fallback IP.

> To determine whether any of this is warranted, I'd need to refresh myself on the purpose(s) for which peerscanner is using the isFallback information.  Was that only so we don't deregister their non-round-robin entries when checks fail for them?

Mainly this is to prevent a situation that we saw occasionally in which a peer would register via using a call proxied via a fallback.